planner/core/point_get_plan.go

```diff
@@ -896,6 +897,36 @@ func tryPointGetPlan(ctx sessionctx.Context, selStmt *ast.SelectStmt, check bool
 	return nil
 }
 
+func pkIsAvailableByHints(idxHints []*ast.IndexHint) bool {
+	return indexIsAvailableByHints(&model.IndexInfo{Name: model.NewCIStr("primary")}, idxHints)
+}
+
+func indexIsAvailableByHints(idxInfo *model.IndexInfo, idxHints []*ast.IndexHint) bool {
+	if len(idxHints) == 0 {
+		return true
+	}
+	for _, hint := range idxHints {
+		if hint.HintScope != ast.HintForScan {
+			continue
+		}
+		if hint.HintType == ast.HintIgnore && hint.IndexNames != nil {
+			for _, name := range hint.IndexNames {
+				if name.L == idxInfo.Name.L {
+					return false
+				}
+			}
+		}
+		if (hint.HintType == ast.HintForce || hint.HintType == ast.HintUse) && hint.IndexNames != nil {
+			for _, name := range hint.IndexNames {
+				if name.L == idxInfo.Name.L {
+					return true
+				}
+			}
+		}
+	}
+	return false
+}
+
```

Comment:
There is a little problem here. If we have unique index `idx_a` and `idx_b` for `select a from t where a = 1 ignore index(idx_b)`, we can not use `point get` because `len(idxHints) != 0` and no `HintForce` `HintUse` for `idx_a`.

Reply:
Oh nice cache. I'll fix it soon.

Reply:
updated